frontend/src/components/Main.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch, batch } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+
+import secrets from '../reducers/secrets'
+import user from '../reducers/user'
+
+import { API_URL } from '../reusable/urls'
+
+const Main = () => {
+  const [newSecret, setNewSecret] = useState('')
+  const accessToken = useSelector(store => store.user.accessToken)
+  const secretItems = useSelector(store => store.secrets.items)
+
+  const history = useHistory()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    if (!accessToken) {
+      history.push('/login')
+    }
+  }, [accessToken, history])
+
+  useEffect(() => {
+    const config = {
+      method: 'GET',
+      headers: {
+        'Authorization': accessToken
+      }
+    }
+
+    fetch(API_URL('secrets'), config)
+      .then(res => res.json())
+      .then(data => {
+        if (data.success) {
+          batch(() => {
+            dispatch(secrets.actions.setSecrets(data.secrets))
+            dispatch(secrets.actions.setErrors(null))
+          })
+        } else {
+          dispatch(secrets.actions.setErrors(data))
+        }
+      })
+  }, [accessToken, dispatch])
```

Comment:
the second useEffect is for the situation when you have accessToken, so why do not input logic from it as the `else` block into the first one? 

```jsx
useEffect(() => {
    if (!accessToken) {
      history.push('/login')
    } else {
      const config = { ... }
      ...
    }
  }, [accessToken, history])
```